core/internal/tensorboard/tfeventreader.go

```diff
@@ -120,22 +121,22 @@ func (s *TFEventReader) NextEvent(
 // It returns whether the buffer has the desired amount of data.
 func (s *TFEventReader) ensureBuffer(
 	count uint64,
-	onNewFile func(string),
+	onNewFile func(paths.AbsolutePath),
 ) bool {
 	if uint64(len(s.buffer)) >= count {
 		return true
 	}
 
 	// If we haven't found the first file, try to find it.
-	if s.currentFile == "" {
+	if s.currentFile == nil {
 		var err error
-		s.currentFile, err = nextTFEventsFile(s.logDir, "", s.fileFilter)
+		s.currentFile, err = nextTFEventsFile(s.logDir, nil, s.fileFilter)
```

Comment:
nit: could you add a comment for the `nil` argument?

Reply:
I created a `nextTFEventsFile` method that takes no arguments instead :)

Reply:
Ah didn’t get there yet… noce